lib/octokit/client/stats.rb

```diff
@@ -75,11 +75,32 @@ def punch_card_stats(repo, options = {})
       #
       # @param repo [Integer, String, Hash, Repository] A GitHub repository
       # @param metric [String] The metrics you are looking for
-      # @return [Array<Sawyer::Resource>] Magical unicorn stats
+      # @return [Array<Sawyer::Resource> or nil] Stats in metric-specific format, or nil if not yet calculated.
+      # @see https://developer.github.com/v3/repos/statistics/
       def get_stats(repo, metric, options = {})
-        data = get("#{Repository.path repo}/stats/#{metric}", options)
+        path = "#{Repository.path repo}/stats/#{metric}"
+        if retry_timeout = options.delete(:retry_timeout)
+          get_stats_with_retry(retry_timeout, path, options)
+        else
+          get(path, options)
+        end.tap do
+          return nil if last_response.status == 202
+        end
+      end
+
+      # @private Get stats for a repository. Retry call until stats are ready or retry_timeout exceeded.
+      #
+      # @param retry_timeout [String] For how long Octokit should keep trying to get stats (in milliseconds)
+      # @return [Array<Sawyer::Resource> or nil] Stats in metric-specific format, or nil if not yet calculated.
+      def get_stats_with_retry(retry_timeout, path, options = {})
+        time_start = Time.now
 
-        last_response.status == 202 ? nil : data
+        loop do
+          data = get(path, options)
+          timeout_exceeded = Time.now - time_start >= retry_timeout
+          return data if last_response.status != 202 || timeout_exceeded
+          sleep retry_timeout / 3.0
```

Comment:
To make sure that there will be no more than two retries. But now I think that three retries will be more suitable, so I've changed denominator to `4`. For better understanding:
```
> Octokit.contributors_stats('octokit/octokit.rb', retry_timeout: 10)
!starting #get_stats_with_retry with retry_timeout: 10 sec!

sending call... (1)
time left: 6.8 sec
- sleeping for 2.5 sec...
sending call... (2)
time left: 0.6 sec
- sleeping for 2.5 sec...
sending call... (3)
time left: -6.7 sec: TIMEOUT EXCEEDED

!stats still not calculated, returning nil!
```